lib/puppet/transaction/report.rb

```diff
@@ -377,9 +377,15 @@ def raw_summary
   #
   def exit_status
     status = 0
-    status |= 2 if @metrics["changes"]["total"] > 0
-    status |= 4 if @metrics["resources"]["failed"] > 0
-    status |= 4 if @metrics["resources"]["failed_to_restart"] > 0
+    if @metrics["changes"] && @metrics["changes"]["total"] &&
+        @metrics["resources"] && @metrics["resources"]["failed"] &&
+        @metrics["resources"]["failed_to_restart"]
+      status |= 2 if @metrics["changes"]["total"] > 0
+      status |= 4 if @metrics["resources"]["failed"] > 0
+      status |= 4 if @metrics["resources"]["failed_to_restart"] > 0
+    else
+      status = -1
```

Comment:
I'm not sure if `nil` would be a more appropriate return, as it is more unknowny, but the api doc suggests Integer is the return type.

Reply:
@adrienthebo in your experience, what would better communicate this error state?

Reply:
@Magisus nominally I'd say 16 or some other somewhat low value; that said we might as well just merge this now as it works as intended.

Reply:
Fine with me